code/game/machinery/computer/modular_shield.dm

```diff
@@ -0,0 +1,82 @@
+/obj/machinery/computer/modular_shield
+	name = "modular shield control console"
+	desc = "Used to remotely monitor and toggle modular shield generators."
+	circuit = /obj/item/circuitboard/computer/modular_shield_console
+
+	///the list of generators that are linked to us
+	var/list/obj/machinery/modular_shield_generator/generators = list()
+
+//lets monkeys randomly mash buttons to toggle the generators
+/obj/machinery/computer/modular_shield/attack_paw(mob/user, list/modifiers)
+	balloon_alert(user, "mashing buttons")
+	if(!do_after(user, 4 SECONDS, target = src))
+		return
+	for(var/obj/machinery/modular_shield_generator/generator in generators)
+		if(prob(50))
+			generator.toggle_shields()
+	return
```

Comment:
```suggestion
```

General note. Return statement at the end of an proc isn't necessary if you aren't returning a value